File: src/backend/BePro.Core/Entities/Candidate.cs
using BePro.Core.Entities.Base;
using BePro.Core.Enums;

namespace BePro.Core.Entities;

public class Candidate : BaseEntity
{
    public string FullName { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public DateTime InterviewDate { get; set; }

    public Guid RecruiterId { get; set; }
    public User Recruiter { get; set; } = null!;

    public Guid LeaderId { get; set; }
    public User Leader { get; set; } = null!;

    public Guid ClientId { get; set; }
    public Client Client { get; set; } = null!;

    public TimeOnly? InterviewTime { get; set; }
    public string? Position { get; set; }
    public string? Municipality { get; set; }
    public int? Age { get; set; }
    public string? Shift { get; set; }
    public string? Plant { get; set; }
    public string? InterviewPoint { get; set; }
    public string? Comments { get; set; }

    public bool Attended { get; set; }
    public CandidateStatus Status { get; set; } = CandidateStatus.Registered;
    public RejectionCategory? RejectionCategory { get; set; }
    public string? RejectionDetails { get; set; }

    public Placement? Placement { get; set; }
}

File: src/frontend/src/types/candidate.ts
export type CandidateStatus =
  | "registered"
  | "interview_scheduled"
  | "attended"
  | "no_show"
  | "pending"
  | "approved"
  | "rejected"
  | "declined"
  | "discarded"
  | "hired"
  | "in_guarantee"
  | "guarantee_met"
  | "guarantee_failed"
  | "replacement";

export type RejectionCategory =
  | "interview"
  | "medical_exam"
  | "background_check"
  | "documentation"
  | "no_show"
  | "salary"
  | "schedule"
  | "transportation"
  | "personal_decision"
  | "other";

export interface ICandidateDto {
  id: string;
  fullName: string;
  phone: string;
  interviewDate: string;
  interviewTime?: string;
  position?: string;
  municipality?: string;
  age?: number;
  shift?: string;
  plant?: string;
  interviewPoint?: string;
  comments?: string;
  attended: boolean;
  status: CandidateStatus;
  rejectionCategory?: RejectionCategory;
  rejectionDetails?: string;
  recruiterId: string;
  recruiterFullName: string;
  leaderId: string;
  leaderFullName: string;
  clientId: string;
  clientName: string;
  createdAt: string;
}

export interface ICreateCandidateRequest {
  fullName: string;
  phone: string;
  interviewDate: string;
  clientId: string;
  interviewTime?: string;
  position?: string;
  municipality?: string;
  age?: number;
  shift?: string;
  plant?: string;
  interviewPoint?: string;
  comments?: string;
}

export interface IUpdateCandidateStatusRequest {
  status: CandidateStatus;
  rejectionCategory?: RejectionCategory;
  rejectionDetails?: string;
}
